zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java

```diff
@@ -0,0 +1,1182 @@
+        mSecondGradientTo = AttrUtils.getColorFromAttr(attrs, "zpb_second_gradient_to", 0xffFF4081);
+        mRadius = AttrUtils.getDimensionFromAttr(attrs, "zpb_round_rect_radius", 20);
+        mDrawBorder = AttrUtils.getBooleanFromAttr(attrs, "zpb_draw_border", false);
+        mBorderWidth = AttrUtils.getDimensionFromAttr(attrs, "zpb_border_width", 1);
+        mBorderColor = AttrUtils.getColorFromAttr(attrs, "zpb_border_color", 0xffff001f);
+    }
+
+    private void initPaths() {
+        mProgressPaint = new Paint();
+        Color hmosColor = ZzHorizontalProgressBar.changeParamToColor(mProgressColor);
+        mProgressPaint.setColor(hmosColor);
+        mProgressPaint.setStyle(Paint.Style.FILL_STYLE);
+        mProgressPaint.setAntiAlias(true);
+
+        mSecondProgressPaint = new Paint();
+        Color hmosColor1 = ZzHorizontalProgressBar.changeParamToColor(mSecondProgressColor);
+        mSecondProgressPaint.setColor(hmosColor1);
+        mSecondProgressPaint.setStyle(Paint.Style.FILL_STYLE);
+        mSecondProgressPaint.setAntiAlias(true);
+
+        mGradientPaint = new Paint();
+        mGradientPaint.setStyle(Paint.Style.FILL_STYLE);
+        mGradientPaint.setAntiAlias(true);
+
+        mSecondGradientPaint = new Paint();
+        mSecondGradientPaint.setStyle(Paint.Style.FILL_STYLE);
+        mSecondGradientPaint.setAntiAlias(true);
+
+        mBgPaint = new Paint();
+        Color hmosColor2 = ZzHorizontalProgressBar.changeParamToColor(mBgColor);
+        mBgPaint.setColor(hmosColor2);
+        mBgPaint.setStyle(Paint.Style.FILL_STYLE);
+        mBgPaint.setAntiAlias(true);
+
+        mBorderPaint = new Paint();
+        Color hmosColor3 = ZzHorizontalProgressBar.changeParamToColor(mBorderColor);
+        mBorderPaint.setColor(hmosColor3);
+        mBorderPaint.setStyle(Paint.Style.STROKE_STYLE);
+        mBorderPaint.setStrokeWidth(mBorderWidth);
+        mBorderPaint.setAntiAlias(true);
+
+    }
+
+    @Override
+    public void onDraw(Component component, Canvas canvas) {
+        switch (mShowMode) {
+            case SHOW_MODE_ROUND:
+                //half circle
+                drawBackgroundCircleMode(canvas);
+                drawProgressCircleMode(canvas);
+                drawBorderCircleMode(canvas);
+                break;
+            case SHOW_MODE_RECT:
+                //rect
+                drawBackgroundRectMode(canvas);
+                drawProgressRectMode(canvas);
+                drawBorderRectMode(canvas);
+                break;
+            case SHOW_MODE_ROUND_RECT:
+                //custom radius
+                drawBackgroundRoundRectMode(canvas);
+                drawProgressRoundRectMode(canvas);
+                drawBorderRoundRect(canvas);
+                break;
+            default:
+                HiLog.debug(hiLogLabel, "default switch case in OnDraw");
+        }
+    }
+
+    /**
+     * 绘制半圆形进度.
```

Comment:
Add the comments in English. Applies to all the methods.

Reply:
done